manage.py

```diff
@@ -562,25 +563,84 @@ def _run_field_lists(name, table, tables_only):
         no_field_list_type = sql.SQL('TABLE' if table.is_table else 'VIEW')
         final_summary_type = sql.SQL('VIEW')
 
+    counts_per_path_select = """
+        SELECT
+            {primary_keys},
+            path,
+            GREATEST(sum(array_item), sum(object_property)) path_count
+        FROM
+            {summary_table}
+        CROSS JOIN
+            flatten({summary_table}.{data_column})
+        GROUP BY
+            {primary_keys}, path
+    """
+
+    # for contracts and award add paths across the awardID join.
+    if summary_table in ('contracts_summary', 'awards_summary'):
+        if summary_table == 'contracts_summary':
+            variables.extend(['awards/', 'awards'])
+        else:
+            variables.extend(['contracts/', 'contract'])
+
+        counts_per_path_select += """
+        UNION ALL
+
+        SELECT
+            {qualified_primary_keys}, %s || path, GREATEST(sum(array_item), sum(object_property)) path_count
+        FROM
+            awards_summary
+        JOIN
+            contracts_summary
+        ON
+            awards_summary.id = contracts_summary.id AND
+            awards_summary.award_id = contracts_summary.awardid
+        CROSS JOIN
+            flatten(contracts_summary.contract)
+        GROUP BY
+            {qualified_primary_keys}, path
+
+        UNION ALL
+
+        SELECT
+            {qualified_primary_keys}, %s as path, count(*) path_count
+        FROM
+            awards_summary
+        JOIN
+            contracts_summary
+        ON
+            awards_summary.id = contracts_summary.id AND
+            awards_summary.award_id = contracts_summary.awardid
+        GROUP BY {qualified_primary_keys}
+    """
+
     # Create a *_field_list table, add a unique index, rename the *_summary table to *_no_field_list, and re-create
     # the *_summary table. Then, copy the comments from the old to the new *_summary table.
-
-    # Use jsonb_object_agg instead of array_agg so that paths are unique, and so that queries against the field use the
-    # faster "in" operator for objects (?&) than for arrays (@>).
     statement = """
         CREATE TABLE {field_list_table} AS
+        WITH path_counts AS (
+            {inner_select}
```

Comment:
To avoid ambiguity, use `INNER_SELECT` as the text to replace. That way, no one can be confused about this being a placeholder for psycopg2 methods.

```suggestion
            INNER_SELECT
```